test/WebSites/MvcWithNullable/Program.cs

```diff
@@ -28,16 +30,39 @@
 [Route("api/[controller]")]
 public class EnumController : ControllerBase
 {
-    [HttpGet]
-    public IActionResult Get(LogLevel? logLevel = LogLevel.Error) => Ok(new { logLevel });
+    [HttpGet("query")]
+    public IActionResult GetQuery(LogLevel? logLevel = LogLevel.Error) => Ok(new { logLevel });
+
+    [HttpGet("path/{logLevel}")]
+    public IActionResult GetPath(LogLevel? logLevel = LogLevel.Error) => Ok(new { logLevel });
+
+    [HttpGet("header")]
+    public IActionResult GetHeader([FromHeader] LogLevel? logLevel = LogLevel.Error) => Ok(new { logLevel });
+
+    [HttpGet("enum-body")]
+    public IActionResult GetEnumBody([FromBody] LogLevel? logLevel = LogLevel.Error) => Ok(new { logLevel });
+
+    [HttpGet("type-body")]
+    public IActionResult GetTypeBody([FromBody] TypeWithNullable typeWithNullable) => Ok(new { typeWithNullable.LogLevel });
 }
 
 [ApiController]
 [Route("api/[controller]")]
 public class RequiredEnumController : ControllerBase
 {
     [HttpGet]
-    public IActionResult Get([Required] LogLevel? logLevel = LogLevel.Error) => Ok(new { logLevel });
+    public IActionResult Get([FromBody, Required] LogLevel? logLevel = LogLevel.Error) => Ok(new { logLevel });
```

Comment:
Why changed to `[FromBody]`?

Reply:
When I fixed outputting null for non body parameters, having a test for a query string parameter no longer made sense as `Required` no longer impacted the result. Whilst it isn't possible to submit a query string with null as the value, `null` is a valid json payload.